crates/integrations/datafusion/src/physical_plan/expr_to_predicate.rs

```diff
@@ -403,4 +419,39 @@ mod tests {
             Reference::new("ts").greater_than_or_equal_to(Datum::string("2023-01-05T00:00:00"));
         assert_eq!(predicate, expected_predicate);
     }
+
+    #[test]
+    fn test_to_timestamp_comparison_creates_predicate() {
+        let sql = "TO_TIMESTAMP(ts) >= timestamp '2023-01-05T00:00:00'";
+        let predicate = convert_to_iceberg_predicate(sql).unwrap();
+        let expected_predicate =
+            Reference::new("ts").greater_than_or_equal_to(Datum::string("2023-01-05T00:00:00"));
+        assert_eq!(predicate, expected_predicate);
+    }
+
+    #[test]
+    fn test_to_timestamp_comparison_to_cast_creates_predicate() {
+        let sql = "TO_TIMESTAMP(ts) >= CAST('2023-01-05T00:00:00' AS TIMESTAMP)";
```

Comment:
What are you trying to achieve in this PR? The following is far easier to solve:
```suggestion
        let sql = "ts >= CAST('2023-01-05T00:00:00' AS TIMESTAMP)";
```

As mentioned in the other comment, we don't want to transform all the values in `ts`, so we want to do something smart there see https://github.com/apache/iceberg-rust/pull/945#issuecomment-2639926317

Reply:
Note that this is `TO_TIMESTAMP`, not `TO_DATE` - so no truncation is supposed to occur, and the current implementation in this PR solves ok I think.